src/Featurizers/Components/GlobalMinimumFrequencyEstimator.h

```diff
@@ -0,0 +1,195 @@
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "FrequencyEstimator.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+static constexpr char const * const         GlobalMinimumFrequencyEstimatorName("GlobalMinimumFrequencyEstimator");
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         GlobalMinimumFrequencyAnnotation
+///  \brief         This is an annotation class which holds the frequency
+///                 of TimeSeries
+///
+class GlobalMinimumFrequencyAnnotation{
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using FrequencyType                     = std::chrono::system_clock::duration;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    FrequencyType                           Value;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    GlobalMinimumFrequencyAnnotation(FrequencyType value);
+    ~GlobalMinimumFrequencyAnnotation(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(GlobalMinimumFrequencyAnnotation);
+};
+
+namespace Details {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         GlobalMinimumFrequencyTrainingOnlyPolicy
+///  \brief         `GlobalMinimumFrequencyEstimator` implementation details.
+///
+template <typename GlobalMinimumFrequencyEstimatorT, size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()>
+class GlobalMinimumFrequencyTrainingOnlyPolicy {
+public:
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using GrainType                     = std::vector<std::string>;
+    using InputType                     = std::tuple<GrainType const &, std::chrono::system_clock::time_point const &>;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const     NameValue = GlobalMinimumFrequencyEstimatorName;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    GlobalMinimumFrequencyTrainingOnlyPolicy(void);

```

Comment:
>GlobalMinimumFrequencyTrainingOnlyPolicy(void); [](start = 4, length = 47)

It doesn't look like this constructor is doing anything and it can be removed.

Reply:
Good catch. I have removed it now.